src/firefly/java/edu/caltech/ipac/firefly/server/visualize/hipslists/IrsaHiPSListSource.java

```diff
@@ -0,0 +1,99 @@
+package edu.caltech.ipac.firefly.server.visualize.hipslists;
+
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.firefly.server.visualize.hipslists.HiPSMasterListEntry.PARAMS;
+import edu.caltech.ipac.firefly.data.ServerParams;
+import edu.caltech.ipac.util.AppProperties;
+import edu.caltech.ipac.util.DataGroup;
+import edu.caltech.ipac.firefly.server.util.DsvToDataGroup;
+import edu.caltech.ipac.util.download.FailedRequestException;
+import edu.caltech.ipac.util.DataObject;
+import edu.caltech.ipac.util.DataType;
+import org.apache.commons.csv.CSVFormat;
+
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Created by cwang on 2/27/18.
+ */
+public class IrsaHiPSListSource implements HiPSMasterListSourceType {
+    private static final Logger.LoggerImpl _log = Logger.getLogger();
+    private final static String irsaHipsTable = "/edu/caltech/ipac/firefly/resources/irsa-hips-master-table.csv";
+    private static final String IRSA_HIPS_TABLE = AppProperties.getProperty("irsa.hipsmastertable.location",irsaHipsTable);
+
+
+    private static Map<String, String> paramsMap = new HashMap<>();
+    static {
+        HiPSMasterListEntry.setParamsMap(paramsMap, PARAMS.TITLE, "label");
+        HiPSMasterListEntry.setParamsMap(paramsMap, PARAMS.URL, "url");
+    }
+
+    public List<HiPSMasterListEntry> getHiPSListData(String[] dataTypes, String source) {
+        try {
+            return createHiPSList(IRSA_HIPS_TABLE, dataTypes, source);
+        }
+        catch (FailedRequestException | IOException e) {
+            _log.info("get Irsa HiPS failed");
+            e.printStackTrace();
+            return null;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+
+    }
+
+    private List<HiPSMasterListEntry> createHiPSList(String hipsMaser,
+                                                     String[] dataTypes,
+                                                     String source) throws IOException, FailedRequestException {
+
+        InputStream inf= IrsaHiPSListSource.class.getResourceAsStream(hipsMaser);
+        DataGroup dg = DsvToDataGroup.parse(inf, CSVFormat.DEFAULT);
+
+        return getListData(dg, paramsMap, source);
+    }
+
+
+    // this is temporary solution for getting HiPS from IRSA
```

Comment:
I just wonder why is this temporary? Is there a plan for something else?

Reply:
Is there some way to get HiPS from IRSA?  For this release to make HiPS source framework to aggregate HiPS from various sources,  a hard code csv file is created and meant to contain the info of HiPS from IRSA.

Reply:
No, IRSA doesn't have a registry service where you can get the list and i don't think is there any plan to do that in the (near) future so the solution you implemented is probably more definitive than temporary ;-)